plutus/budget_manager/budget_manager.py

```diff
@@ -0,0 +1,230 @@
+import yaml
+
+from google.cloud import billing_budgets_v1beta1
+from google.cloud import resource_manager
+from google.cloud import storage
+
+from google.api_core.exceptions import GoogleAPICallError, RetryError
+
+from constants import (
+    APP,
+    PLUTUS_CONFIG_TYPE_PROJECT,
+    PLUTUS_CONFIG_TYPE_PARENT,
+    PLUTUS_CONFIG_TYPE_LABEL,
+    PLUTUS_CONFIG_TYPE_DEFAULT
+)
+
+from mysql import upsert_budget
+from gcp_helper import GcpHelper
+from project_budget import ProjectBudget
+
+# TODO - CircleCI should push yaml config to GCS after master branch pushes
+# TODO - add sample config yaml file to repo
+
+log = logging.getLogger(APP)
+metrics = markus.get_metrics(APP)
+
+@click.command()
+@click.option('--gcs-bucket', default=None)
+@click.option('--gcs-file-path', default=None)
+@click.option('--billing-account-id', default=None)
+@click.option('--default-pubsub-topic', default='projects/moz-fx-billing-212017/topics/plutus-budget-notifications')
+@click.option('--mysql-host', default='localhost')
+@click.option('--mysql-port', default='3306')
+@click.option('--mysql-user', default='root')
+@click.option('--mysql-pass', default='secret')
+@click.option('--mysql-db', default='plutus')
+# Used for local testing without container, and without using gcs file
+@click.option('--local-mode', is_flag=True)
+@click.option('--dry-run', is_flag=True)
+def main(gcs_bucket, gcs_file_path, billing_account_id, default_pubsub_topic, mysql_host, mysql_port, mysql_user, mysql_pass, mysql_db, local_mode, dry_run):
+    setup_logging(APP)
+    setup_metrics(APP)
+
+    # Load config.yaml file
+    if local_mode:
+        with open("config.yaml", "r") as f:
+            budget_dict = yaml.load(f, Loader=yaml.SafeLoader)
+    else:
+        # Load yaml configuration file from GCS
+        gcs_client = storage.Client()
+        bucket = gcs_client.bucket(gcs_bucket)
+        blob = bucket.blob(gcs_file_path)
+        blob.download_to_filename('/app/gcs_budget_config.yaml')
+        with open('/app/gcs_budget_config.yaml') as f:
+            budget_dict = yaml.load(f, Loader=yaml.SafeLoader)
+
+    # Setup gcp helper
+    billing_client = billing_budgets_v1beta1.BudgetServiceClient()
+    rm_client = resource_manager.Client()
+    gcp = GcpHelper(billing_client, rm_client)
+
+    # Setup mysql connection pool
+    pool = PooledDB(creator = pymysql,
+                    host = mysql_host,
+                    user = mysql_user,
+                    password = mysql_pass,
+                    database = mysql_db,
+                    autocommit = True,
+                    blocking = True,
+                    maxconnections = 5)
+    mysql_conn = pool.connection()
+
+    # Iterate over all configured project budgets
+    for project_dict in budget_dict['projects']:
+        config_type = PLUTUS_CONFIG_TYPE_PROJECT
+
+        if not verify_project_yaml(project_dict):
```

Comment:
As part of code flow I think `if verify_project_yaml(project_dict): ... with positive conditions else: failure conditions` would be better for code readability.

Reply:
i thought the original way was more readable, fail fast and less indentation to parse. But ive changed it to recommended anyway